### PR TITLE
refactor: update rrule-temporal and remove Google UNTIL Workaround

### PR DESCRIPTION
**Pull request by KristjanESPERANTO**
This PR updates the functional RRULE processing to `rrule-temporal@1.4.6` and removes the previously local workaround for Google Calendar's DATE-only `UNTIL` with DATE-TIME `DTSTART`.

## Changes
- Dependency update to `rrule-temporal@1.4.6`
- Removes the local "Google UNTIL workaround" in `ical.js`
- Adds test documentation in `test/google-calendar-until-bug.test.js` clarifying that responsibility now lies with `rrule-temporal`

## Workaround History
- The removed workaround was only recently added: 1155e5a9f3fba7ed996376353b0c741630bc0992
  - Commit: `fix: handle Google Calendar UNTIL without time respecting event timezone (#436)`
- This PR removes the workaround again, as the fix is now handled upstream in `rrule-temporal@1.4.6`

## Benefits
- **Less special-case logic in parser:** 28 lines of workaround code eliminated
- **Clearer responsibilities:** UNTIL normalization belongs with the library that actually interprets RRULE
- **Reduced maintenance burden:** less corner-case code to maintain in `node-ical`
- **Better traceability:** test file explicitly documents the Google case and expected interpretation

## Risk / Impact
- Functional change intentionally depends on `rrule-temporal@1.4.6`
- No API breaking change in `node-ical`; behavior remains protected by regression tests

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved handling of recurring event end dates with UNTIL values.

* **Chores**
  * Updated rrule-temporal dependency to v1.4.6.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,7 +21,7 @@
     "node": ">=18"
   },
   "dependencies": {
-    "rrule-temporal": "^1.4.5"
+    "rrule-temporal": "^1.4.6"
```
[the remark this answers is not in the document]
Comment (KristjanESPERANTO) [reply package.json:24]: It was published yesterday: https://github.com/ggaabe/rrule-temporal/releases/tag/v1.4.6
